### PR TITLE
docs(software-templates): updating docs for required vars in software-templates

### PR DESCRIPTION
**Pull request by benjdlambert**
These whole docs will be getting re-written soon. But let's just update this thing for now.

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4334-lrydaz94jnx7goahsfh6iofow8ukivdz.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6017d09e353bbb25c99c7a80

File: docs/features/software-templates/index.md [hunk 1]
```diff
@@ -40,10 +40,9 @@ internally.
 
 After filling in these variables, you'll get some more fields to fill out which
 are required for Backstage usage: the owner (which is a `user` in the backstage
-system), the `storePath` (which right now must be a GitHub Organisation or
-GitHub user and a non-existing GitHub repository name in the format
-`organisation/reponame`), and a GitHub team or user account which should be
-granted admin access to the repository.
+system), and the `storePath` which is a destination URL to create for the
+provider for instance http://github.com/backstage/my-new-repository, or
```
Comment (freben) [inline docs/features/software-templates/index.md:44]:
```suggestion
provider, for instance `https://github.com/backstage/my-new-repository`, or
```

File: docs/features/software-templates/index.md [hunk 2]
```diff
@@ -40,10 +40,9 @@ internally.
...
+https://gitlab.com/myorg/myrepo.
```
Comment (freben) [inline docs/features/software-templates/index.md:45]:
```suggestion
`https://gitlab.com/myorg/myrepo`.
```

**Review by freben — APPROVED** [review]
two nits, but pre-approved anyway

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4334-lrydaz94jnx7goahsfh6iofow8ukivdz.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6017d09e353bbb25c99c7a80

**Pull request merged**